qiskit/visualization/dag/dagstyle.py

```diff
@@ -0,0 +1,211 @@
+        opnodefontcolor (str): The font color to use for Instruction nodes.
+            Overrides nodecolor for those nodes.
+
+        qubitedgecolor (str): The edge color for qubits. Overrides edgecolor for these edges.
+        clbitedgecolor (str): The edge color for clbits. Overrides edgecolor for these edges.
+    """
+
+    def __init__(self):
+        default_style_dict = "color.json"
+        path = Path(__file__).parent / "styles" / default_style_dict
+
+        with open(path, "r") as infile:
+            default_style = json.load(infile)
+
+        # set shortcuts, such as "ec" for "edgecolor"
+        self.style = StyleDict(**default_style)
+
+
+def load_style(style: dict | str = "color") -> StyleDict:
+    """Utility function to load style from json files.
+
+    Args:
+        style: Depending on the type, this acts differently:
+
+            * If a string, it can specify a supported style name (such
+              as "iqp" or "clifford"). It can also specify the name of
+              a custom color scheme stored as JSON file. This JSON file
+              _must_ specify a complete set of colors.
+            * If a dictionary, it may specify the style name via a
+              ``{"name": "<desired style>"}`` entry. If this is not given,
+              the default style will be used. The remaining entries in the
+              dictionary can be used to override certain specs.
+              E.g. ``{"name": "iqp", "ec": "#FF0000"}`` will use the ``"iqp"``
+              color scheme but set the edgecolor to red.
+
+    Returns:
+        A tuple containing the style as dictionary and the default font ratio.
+    """
+
+    # determine the style name which could also be inside a dictionary, like
+    # style={"name": "clifford", <other settings...>}
+    if isinstance(style, dict):
+        style_name = style.get("name", "color")
+    elif isinstance(style, str):
+        if style.endswith(".json"):
+            style_name = style[:-5]
+        else:
+            style_name = style
+    else:
+        raise exceptions.VisualizationError(f"Invalid style {style}")
+
+    if style_name == "color":
+        current_style = DefaultStyle().style
+    else:
+        # Search for file in 'styles' dir, and then the current directory
+        style_name = style_name + ".json"
+        style_paths = []
+
+        default_path = Path(__file__).parent / "styles" / style_name
+        style_paths.append(default_path)
+
+        # check current directory
+        cwd_path = Path("") / style_name
+        style_paths.append(cwd_path)
+
+        for path in style_paths:
+            # expand ~ to the user directory and check if the file exists
+            exp_user = path.expanduser()
+            if os.path.isfile(exp_user):
+                try:
+                    with open(exp_user) as infile:
+                        json_style = json.load(infile)
+
+                    current_style = StyleDict(json_style)
```

Comment:
This is the case where it might be a little difficult to reuse the `StyleDict` object if we plan on using `load_style` from `visualization/qc_style.py`. Perhaps we could create an inner method that is called by both this and `qc_style.py` `load_style` in which you can specify the type of `StyleDict` being used. But again, that might be a bit tricky with Python, though not impossible.